### PR TITLE
Adds Charged Cries to GNS

### PR DESCRIPTION
**Pull request by DatBoiTim**
Adds charged Cries

## About The Pull Request

Adds "charged" cries to the GNS. These cries will randomly occur instead of the default cry after doing a specific number of chained punches. The charged cry is automatically brought up in the war-cry change dialogue after the initial cry is determined.

Currently, for a punch to be considered "chained" it must be less than half a second from the previous.

There are no statistical changes to the punch based on chain length, only the war-cry changes.

Note: Using *flip and *spin for warcry/charged warcry(or vice versa) will not work. This is due to the function of the emotes themselves, and not the gloves. *flip or *spin for warcry and any audible/text emote will work just fine(or vice versa). No issues unexpected behavior occurs beyond what is defined here.

I will be adding charged warcries to other items/mobs which have such things later. Functional proof of concept.

## Why It's Good For The Game

Allows the GNS to be more accurate to Kenshiro's battlecry from Fist of the Northstar. Users can now get more creative with war cries.

## Changelog
:cl: DatBoiTim
add: Added charged warcries to the Gloves of the Northstar.
add: Added a method to change the charged warcry.
/:cl:

### CODE REVIEW

**Comment by MNarath1** [comment]
how about making the charged warcry also customable so that you can decide if you want to go for the reference or not

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 1]
```diff
@@ -63,21 +63,62 @@
...
-				M.say("[warcry]", ignore_spam = TRUE, forced = "north star warcry")
-			
+				if(newClickTime - oldClickTime > chargeMaxDelayTime)
+					M.say("[warcry]", ignore_spam = TRUE, forced = "north star warcry")
+					oldClickTime = newClickTime
+					chainLength = 0
+				else if(chainLength < chargeChainTrigger)
+					M.say("[warcry]", ignore_spam = TRUE, forced = "north star warcry")
+					oldClickTime = newClickTime
+					chainLength = chainLength + 1
```
Comment (PowerfulBacon) [inline code/modules/clothing/gloves/miscellaneous.dm]: use chainLength ++ for these

**Review by PowerfulBacon — COMMENTED** [review]
I guess it works, but whats the point?

**Comment by DatBoiTim** [comment]
> I guess it works, but whats the point?

Makes gloves a more in depth reference to the source content. Players can get even more creative with other references and jokes.

I'm eventually gonna add similar to guardians as well but this is a test to see the reception.

File: code/modules/clothing/gloves/miscellaneous.dm (hunk 1 above)
Comment (DatBoiTim) [reply code/modules/clothing/gloves/miscellaneous.dm]: Resolved by commit 6d57315

**Comment by ike709** [comment]
Lotta bloat just to expand upon an existing reference.

**Pull request closed without merging**